File: my_py_pkg/my_py_pkg/add_two_ints_server.py
import rclpy # ROS library
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts

class AddTowIntsServerNode(Node): 
    def __init__(self): 
        super().__init__("add_two_ints_server")
        self.server = self.create_service(AddTwoInts, "add_two_ints")
       


def main(args=None): 
    rclpy.init(args=args) 
    node = AddTowIntsServerNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__": 
    main()

File: my_cpp_pkg/src/add_two_ints_server.cpp
#include <rclcpp/rclcpp.hpp> 
#include <example_interfaces/srv/add_two_ints.hpp> 


class AddTwoIntsServerNode : public rclcpp::Node 
{ 
    public: 
        AddTwoIntsServerNode() : Node("add_two_ints_server")
        {
            server = this->create_service<example_interfaces::srv::AddTwoInts>(
                "add_two_ints", 
                std::bind(&AddTwoIntsServerNode::CallbackAddTwoInts, this, std::placeholders::_1, std::placeholders::_2)
            );

            RCLCPP_INFO(this->get_logger(), "server start");
        }

    private: 
        void CallbackAddTwoInts(example_interfaces::srv::AddTwoInts::Request::SharedPtr request, 
                                example_interfaces::srv::AddTwoInts::Response::SharedPtr response) 
        { 
            response->sum = request->a + request->b; 
            RCLCPP_INFO(this->get_logger(), "%d + %d = %d", request->a, request->b, response->sum);
        }

        rclcpp::Service<example_interfaces::srv::AddTwoInts>::SharedPtr server;

}; 

int main(int argc, char** argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<AddTwoIntsServerNode>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}